engine/src/resource/resource.h

```diff
@@ -37,7 +37,7 @@ class Resource {
     T adjusted_max_value_;
     bool no_max_;
 public:
-    Resource(const T &value, const T &min_value = 0, const T &max_value = -1);
+    Resource(const T &value = 0, const T &min_value = 0, const T &max_value = -1);
```

Comment:
If this constructor can be called with no arguments, or only one argument, it might be best to mark the constructor as `explicit`.